src/event_manager/subscribers.rs

```diff
@@ -0,0 +1,51 @@
+use super::SubscriberId;
+use std::collections::HashMap;
+
+// Internal structure used to keep the set of subscribers registered with an EventManger.
+// This structure is a thin wrapper over a `HashMap` in which the keys are uniquely
+// generated when calling `add`.
+pub(crate) struct Subscribers<T> {
+    // The key is the unique id of the subscriber and the entry is the `Subscriber`.
+    subscribers: HashMap<SubscriberId, T>,
+    // We are generating the unique ids by incrementing this counter for each added subscriber,
+    // and rely on the extremely large value range of u128 to ensure each value is effectively
+    // unique over the runtime of any VMM.
+    next_id: u128,
+}
+
+impl<T> Subscribers<T> {
+    pub(crate) fn new() -> Self {
+        Subscribers {
+            subscribers: HashMap::new(),
+            next_id: 0,
```

Comment:
How about reserving subscriber_id 0?

Reply:
Do you see a problem with starting from 0?

Reply:
No really problem, just want to reserve id 0 for the RemoteEndpoint fd.
And reserving id 0 may help to debug in some cases:)

Reply:
I'll update it.